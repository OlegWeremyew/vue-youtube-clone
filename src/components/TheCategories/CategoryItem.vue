<template>
  <a
      href="#"
      :class="classes"
  >
    {{ category }}
  </a>
</template>

<script>
export default {
  name: "CategoryItem",
  props: {
    category: {
      type: String,
      required: true,
    },
    isActive: {
      type: Boolean,
      required: true,
    }
  },
  computed: {
    classes() {
      const classes = ['px-3', 'py-1', 'transition', 'rounded-full', 'border',]

      return this.isActive
          ? [
            'border-gray-700',
            ...classes,
            'bg-gray-600',
            'hover:bg-gray-500',
            'text-white'
          ]
          : [
            ...classes,
            'border-gray-300',
            'bg-gray-100',
            'hover:bg-gray-200'
          ]
    }
  }
}
</script>

<style scoped>

</style>