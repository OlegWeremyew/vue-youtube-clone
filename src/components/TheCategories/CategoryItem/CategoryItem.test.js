import {render, screen} from '@testing-library/vue';
import CategoryItem from "./CategoryItem.vue";

const category = 'Category'

function renderCategoryItem(isActive = false) {
  const options = {
    props: {
      category,
      isActive,
    },
  }
  return render(CategoryItem, options)
}

it('renders with label', () => {
  renderCategoryItem()

  expect(screen.getByText(category)).not.toHaveClass('bg-gray-600 hover:bg-gray-500 text-white')
})

it('renders active', () => {
  renderCategoryItem(true)

  expect(screen.getByText(category)).toHaveClass('bg-gray-600 hover:bg-gray-500 text-white')
})

it('renders correctly', () => {
  const {html} = renderCategoryItem()

  expect(html()).toMatchSnapshot()
})