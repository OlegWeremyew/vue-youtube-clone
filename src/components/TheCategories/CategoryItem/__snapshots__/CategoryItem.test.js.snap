// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<a href=\\"#\\" class=\\"px-3 py-1 transition rounded-full border border-gray-300 bg-gray-100 hover:bg-gray-200\\">Category</a>"`;
